tests/test_simulation.py

```diff
@@ -934,17 +911,58 @@ def test_run_xy():
     assert sim.samples_obj._measurement == "XY"
 
 
-def test_noisy_xy():
+@pytest.mark.parametrize(
+    "masked_qubit",
```

Comment:
It would make more sense to split this into two variables:
```suggestion
    "masked_qubit, result",
```

Reply:
Thanks for reminding me of this 😅